src/binary-encoding.adoc

```diff
@@ -2298,16 +2298,302 @@ considered "shapes" rather than types because while their encoding is always sta
+
+.Figure {counter:figure}: Definition of macro `_foo_` at address `_0_`
+[%unbreakable,source]
+----
+(macro
+    foo          // Macro name
+    [(x [int])]  // Parameters; `x` is a grouped parameter
+    /*...*/      // Body (elided)
+)
+----
+
+[#length_prefixed_expression_groups]
+==== Length-prefixed Expression Groups
+
+If a grouped parameter's <<argument_encoding_bitmap, AEB bits>> are `0b10`, then the argument expressions belonging
+to that parameter will be prefixed by a `FlexUInt` indicating the number of bytes used to encode them.
+
+.Figure {counter:figure}: Length-prefixed encoding of `_(:foo [1, 2, 3])_`
+[%unbreakable,source]
+----
+┌──── The opcode is less than 0x40, so it is an E-expression invoking the macro at
+│     address 0: `foo`. `foo` takes a group of int expressions as a parameter (`x`),
+│     so an argument encoding bitmap (AEB) follows.
+│  ┌──── AEB: 0b0000_0010; the arguments for grouped parameter `x` have been encoded
+│  │     as a length-prefixed expression group. A FlexUInt length prefix follows.
+│  │  ┌──── FlexUInt: 6; the next 6 bytes are an `int` expression group.
+│  │  │
+00 02 0D 51 01 51 02 51 03
+         └─┬─┘ └─┬─┘ └─┬─┘
+           1     2     3
+----
+
+[#delimited_expression_groups]
+==== Delimited Expression Groups
 
-TODO
+If a grouped parameter's <<argument_encoding_bitmap, AEB bits>> are `0b11`, then the argument expressions belonging
+to that parameter will be encoded in a delimited sequence.
+Delimited sequences are encoded differently for <<tagged_encodings,tagged types>> and
+<<tagless_encodings, tagless types>>.
 
-=== Voidable Parameter Encodings
+===== Delimited Tagged Expression Groups
 
-TODO
+Tagged type encodings begin with an <<opcodes, opcode>>; a delimited sequence of tagged arguments is terminated by
+the closing delimiter opcode, `0xF0`.
+
+.Figure {counter:figure}: Delimited encoding of `_(:foo [1, 2, 3])_`
+[%unbreakable,source]
+----
+┌──── The opcode is less than 0x40, so it is an E-expression invoking the macro at
+│     address 0: `foo`. `foo` takes a group of int expressions as a parameter (`x`),
+│     so an argument encoding bitmap (AEB) follows.
+│  ┌──── AEB: 0b0000_0011; the arguments for grouped parameter `x` have been encoded
+│  │     as a delimited expression group. A series of tagged `int` expressions follow.
+│  │                    ┌──── Opcode 0xF0 ends the expression group.
+│  │                    │
+00 03 51 01 51 02 51 03 F0
+      └─┬─┘ └─┬─┘ └─┬─┘
+        1     2     3
+----
+
+===== Delimited Tagless Expression Groups
+
+Tagless type encodings do not have an opcode, and so cannot use the closing delimiter opcode--`0xF0` is a valid first
+byte for many tagless encodings.
+
+Instead, tagless expressions are grouped into 'pages', each of which is prefixed by a <<flexuint, `FlexUInt`>>
+representing a count (not a byte-length) of the expressions that follow. If a prefix has a count of zero, that marks
+the end of the sequence of pages.
```

Comment:
I'm really quite happy with this approach!  It's super flexible.   The only strong critique I have is that it has high overhead for small sequences of narrow values like `int8`.   But I suppose in those case length-prefixing would be preferred.

Reply:
>  The only strong critique I have is that it has high overhead for small sequences of narrow values like int8. But I suppose in those case length-prefixing would be preferred.

Yes, I think that's true. Writers also have the option of doing a delimited sequence comprised of very few pages (1?) with a large count. The data size difference is likely negligible, though.